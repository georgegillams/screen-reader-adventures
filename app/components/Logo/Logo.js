import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import BpkText from 'bpk-component-text';
import { cssModules } from 'bpk-react-utils';

import STYLES from './logo.scss';

const getClassName = cssModules(STYLES); // REGEX_REPLACED

const Logo = props => {
  const {
    noPadding,
    animated,
    small,
    className,
    alwaysCentered,
    light,
    ...rest
  } = props;
  const classNameFinal = [getClassName('logo__container')];
  if (className) {
    classNameFinal.push(className);
  }
  if (alwaysCentered) {
    classNameFinal.push(getClassName('logo__container--centered'));
  }
  if (animated) {
    classNameFinal.push(getClassName('logo__container--animated'));
  }

  const baseTextClassNameFinal = [getClassName('logo__logo-base')];
  if (light) {
    baseTextClassNameFinal.push(getClassName('logo__logo-base--light'));
  }

  const largeTextClassNameFinal = [getClassName('logo__logo-large')];
  if (small) {
    largeTextClassNameFinal.push(getClassName('logo__logo-large--smaller'));
  }
  if (noPadding) {
    classNameFinal.push(getClassName('logo__container--no-padding'));
    largeTextClassNameFinal.push(getClassName('logo__logo-large--no-padding'));
    baseTextClassNameFinal.push(getClassName('logo__logo-base--no-padding'));
  }

  return (
    <div className={classNameFinal.join(' ')} {...rest}>
      <Link role="button" aria-label="Home page" to="/">
        <BpkText textStyle="xxl" className={largeTextClassNameFinal.join(' ')}>
          {'<G/>'}
        </BpkText>
        {!small && (
          <div>
            <br />
            <br />
            <BpkText
              textStyle="lg"
              className={baseTextClassNameFinal.join(' ')}
            >
              {'George Gillams'}
            </BpkText>
          </div>
        )}
      </Link>
    </div>
  );
};

Logo.propTypes = {
  className: PropTypes.string,
  noPadding: PropTypes.bool,
  animated: PropTypes.bool,
  small: PropTypes.bool,
  light: PropTypes.bool,
  alwaysCentered: PropTypes.bool,
};

Logo.defaultProps = {
  className: null,
  noPadding: false,
  animated: false,
  small: false,
  light: false,
  alwaysCentered: false,
};

export default Logo;
