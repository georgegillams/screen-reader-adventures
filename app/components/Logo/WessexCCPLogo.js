import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import BpkText from 'bpk-component-text';
import BpkImage from 'bpk-component-image';

import STYLES from './wessex-ccp-logo.scss';
import { cssModules } from 'bpk-react-utils';
const getClassName = cssModules(STYLES); // REGEX_REPLACED

const LONG_PRESS_DURATION = 3000;

class WessexCCPLogo extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  onMouseLeave = () => {};

  render() {
    const {
      noPadding,
      animated,
      small,
      className,
      alwaysCentered,
      light,
      ...rest
    } = this.props;
    const classNameFinal = [getClassName('wessex-ccp-logo__container')];
    if (className) {
      classNameFinal.push(className);
    }
    if (alwaysCentered) {
      classNameFinal.push(getClassName('wessex-ccp-logo__container--centered'));
    }
    if (animated) {
      classNameFinal.push(getClassName('wessex-ccp-logo__container--animated'));
    }

    const baseTextClassNameFinal = [getClassName('wessex-ccp-logo__logo-base')];
    if (light) {
      baseTextClassNameFinal.push(
        getClassName('wessex-ccp-logo__logo-base--light'),
      );
    }

    const largeTextClassNameFinal = [
      getClassName('wessex-ccp-logo__logo-large'),
    ];
    if (small) {
      largeTextClassNameFinal.push(
        getClassName('wessex-ccp-logo__logo-large--smaller'),
      );
    }
    if (noPadding) {
      classNameFinal.push(
        getClassName('wessex-ccp-logo__container--no-padding'),
      );
      largeTextClassNameFinal.push(
        getClassName('wessex-ccp-logo__logo-large--no-padding'),
      );
      baseTextClassNameFinal.push(
        getClassName('wessex-ccp-logo__logo-base--no-padding'),
      );
    }

    return (
      <div className={classNameFinal.join(' ')} {...rest}>
        <a
          role="button"
          aria-label="Main conference page"
          onMouseEnter={() => {
            this.setState({ hovering: true });
          }}
          onFocus={() => {
            this.setState({ hovering: true });
          }}
          onMouseLeave={() => {
            this.setState({ hovering: false });
          }}
          onBlur={() => {
            this.setState({ hovering: false });
          }}
          href="http://www.epicc-conference.org/"
          onMouseLeave={this.onMouseLeave}
        >
          <BpkImage
            src={
              small
                ? 'https://i.imgur.com/h5zmNwn.png'
                : 'https://i.imgur.com/3nVx6yS.png'
            }
            className={largeTextClassNameFinal.join(' ')}
            width={2000}
            height={small ? 348 : 187}
          />
        </a>
      </div>
    );
  }
}

WessexCCPLogo.propTypes = {
  className: PropTypes.string,
  noPadding: PropTypes.bool,
  animated: PropTypes.bool,
  small: PropTypes.bool,
  light: PropTypes.bool,
  alwaysCentered: PropTypes.bool,
};

WessexCCPLogo.defaultProps = {
  className: null,
  noPadding: false,
  animated: false,
  small: false,
  light: false,
  alwaysCentered: false,
};

export default WessexCCPLogo;
