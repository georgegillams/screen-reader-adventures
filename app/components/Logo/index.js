export Logo from './Logo';
export WessexCCPLogo from './WessexCCPLogo';
