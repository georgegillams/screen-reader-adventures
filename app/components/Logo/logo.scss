@import '~utils/tokens';

@fontface {
  font-family: 'Bungee Inline', cursive;
  src: url('https://fonts.googleapis.com/css?family=Bungee+Inline');
}

@fontface {
  font-family: 'Neuton', serif;
  src: url('https://fonts.googleapis.com/css?family=Neuton');
}

@fontface {
  font-family: 'Quattrocento Sans', sans-serif;
  src: url('https://fonts.googleapis.com/css?family=Quattrocento+Sans');
}

@fontface {
  font-family: 'Iceland', cursive;
  src: url('https://fonts.googleapis.com/css?family=Iceland');
}

@keyframes example {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.1);
  }
  100% {
    transform: scale(1);
  }
}

.logo {
  &__container {
    display: flex;
    padding-bottom: 1rem;
    flex-direction: column;
    transition: all 0.2s;

    @include bpk-breakpoint-tablet {
      padding-right: 2rem;
      padding-left: 2rem;
      align-items: center;
    }

    &--centered {
      align-items: center;
    }

    &--no-padding {
      padding: 0;
    }

    &--animated {
      &:focus,
      &:active,
      &:hover {
        animation-name: example;
        animation-duration: 0.4s;
      }
    }
  }

  &__logo-base {
    display: inline-block;
    color: #1e1e1e;
    font-family: 'Quattrocento Sans', sans-serif;
    font-size: 1.6rem;

    &--light {
      color: $bpk-color-sky-gray-tint-06;
    }

    &--no-padding {
      line-height: normal;
    }
  }

  &__logo-large {
    display: inline-block;
    color: $primary-color;
    font-family: 'Quattrocento Sans', sans-serif;
    font-size: 5.5rem;
    font-weight: bold;
    line-height: 6rem;

    @media (prefers-color-scheme: dark) {
      color: $primary-color-invert;
    }

    &--smaller {
      font-size: 3.5rem;

      @include bpk-breakpoint-mobile {
        font-size: 3rem;
      }
    }

    &--no-padding {
      line-height: normal;
    }
  }
}
