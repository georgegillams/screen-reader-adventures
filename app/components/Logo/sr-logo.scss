@import '~bpk-mixins';

@fontface {
  font-family: 'Bungee Inline', cursive;
  src: url('https://fonts.googleapis.com/css?family=Bungee+Inline');
}

@fontface {
  font-family: 'Neuton', serif;
  src: url('https://fonts.googleapis.com/css?family=Neuton');
}

@fontface {
  font-family: 'Quattrocento Sans', sans-serif;
  src: url('https://fonts.googleapis.com/css?family=Quattrocento+Sans');
}

@fontface {
  font-family: 'Iceland', cursive;
  src: url('https://fonts.googleapis.com/css?family=Iceland');
}

@keyframes example {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.1);
  }
  100% {
    transform: scale(1);
  }
}

.sr-logo {
  &__container {
    display: flex;
    padding-bottom: 1rem;
    flex-direction: column;
    transition: all 0.2s;

    @include bpk-breakpoint-tablet {
      padding-right: 2rem;
      padding-left: 2rem;
      align-items: center;
    }

    &--centered {
      align-items: center;
    }

    &--no-padding {
      padding: 0;
    }

    &--animated {
      &:focus,
      &:active,
      &:hover {
        animation-name: example;
        animation-duration: 0.4s;
      }
    }
  }

  &__logo-base {
    display: inline-block;
    color: #1e1e1e;
    font-family: 'Quattrocento Sans', sans-serif;
    font-size: 1.6rem;

    &--light {
      color: $bpk-color-gray-100;
    }

    &--no-padding {
      line-height: normal;
    }
  }

  &__logo-large {
    display: inline-block;
    color: rgb(62, 158, 235);
    font-family: 'Quattrocento Sans', sans-serif;
    font-size: 5.5rem;
    font-weight: bold;
    line-height: 6rem;
    width: 70vw;
    max-width: 40rem;

    &--smaller {
      font-size: 3.5rem;
      width: 18rem;
      padding: 1rem 0;

      @include bpk-breakpoint-tablet {
        width: 12rem;
      }
      @include bpk-breakpoint-mobile {
        width: 9rem;
      }
    }

    &--no-padding {
      line-height: normal;
    }
  }
}
