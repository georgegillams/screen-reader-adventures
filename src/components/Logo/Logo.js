import React from 'react';
import PropTypes from 'prop-types';
import Link from 'next/link';
import { cssModules } from 'gg-components/helpers/cssModules';

import STYLES from './logo.scss';

const getClassName = cssModules(STYLES);

const Logo = props => {
  const { padding, animated, className, alwaysCentred, pride, ...rest } = props;
  const classNameFinal = [getClassName('logo__container')];
  if (className) {
    classNameFinal.push(className);
  }
  if (alwaysCentred) {
    classNameFinal.push(getClassName('logo__container--centred'));
  }

  const largeTextClassNameFinal = [getClassName('logo__heading')];
  if (animated) {
    largeTextClassNameFinal.push(getClassName('logo__subheading--animated'));
  }
  if (pride) {
    largeTextClassNameFinal.push(getClassName('logo__heading--pride'));
  }

  if (!padding) {
    classNameFinal.push(getClassName('logo__container--no-padding'));
    largeTextClassNameFinal.push(getClassName('logo__heading--no-padding'));
  }

  return (
    <div className={classNameFinal.join(' ')} {...rest}>
      <Link href="/">
        <a href="/" className={getClassName('logo__a')}>
          <h1 role="button" aria-label="Home page" className={largeTextClassNameFinal.join(' ')}>
            {'LOGO'}
          </h1>
        </a>
      </Link>
    </div>
  );
};

Logo.propTypes = {
  className: PropTypes.string,
  padding: PropTypes.bool,
  animated: PropTypes.bool,
  pride: PropTypes.bool,
  alwaysCentred: PropTypes.bool,
};

Logo.defaultProps = {
  className: null,
  padding: true,
  animated: false,
  pride: false,
  alwaysCentred: false,
};

export default Logo;
