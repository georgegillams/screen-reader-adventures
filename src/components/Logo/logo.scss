@import '~utils/tokens';

@keyframes example {
  0% {
    transform: scale(1);
  }
  50% {
    transform: scale(1.1);
  }
  100% {
    transform: scale(1);
  }
}

.logo {
  &__container {
    display: flex;
    padding-bottom: $spacing-base;
    flex-direction: column;
    transition: transform 0.2s;
    border-radius: 0.25rem;
    outline: none;

    @include bpk-breakpoint-tablet {
      padding-right: $spacing-lg;
      padding-left: $spacing-lg;
      align-items: center;
    }

    &--centred {
      align-items: center;
    }

    &--no-padding {
      padding: 0;
    }
  }

  &__heading {
    display: inline-block;
    padding-top: 0.1rem;
    border-radius: 0.25rem;
    color: $primary-color;
    font-size: 3rem;
    font-weight: bold;
    line-height: 6rem;

    @include bpk-breakpoint-mobile {
      font-size: 2.6rem;
    }

    @media (prefers-color-scheme: dark) {
      color: $primary-color-invert;
    }

    &--pride {
      @supports (background-clip: text) {
        background-image: linear-gradient(
          to left,
          rgb(136, 65, 136),
          rgb(126, 60, 173),
          rgb(56, 56, 192),
          rgb(47, 177, 47),
          rgb(216, 216, 0),
          rgb(209, 136, 1),
          rgb(209, 28, 28)
        );
        color: transparent;
        background-clip: text;

        @media (prefers-color-scheme: dark) {
          background-image: linear-gradient(
            to left,
            rgb(136, 57, 136),
            rgb(94, 25, 143),
            rgb(57, 57, 161),
            rgb(60, 134, 60),
            rgb(165, 165, 69),
            rgb(175, 140, 75),
            rgb(155, 72, 72)
          );
        }
      }
    }

    &--no-padding {
      margin: 0;
      line-height: normal;
    }
  }

  &__a {
    border-radius: 0.25rem;
    overflow: hidden;

    @include focus-style;

    &:hover,
    &:active,
    &:focus {
      animation-duration: 0.4s;
      animation-name: gg-bounce-hover-effect;
    }
  }
}
