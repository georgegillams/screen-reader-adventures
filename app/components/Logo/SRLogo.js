import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router';
import BpkText from 'bpk-component-text';
import BpkImage from 'bpk-component-image';
import Logo from './screen_reader_adventures.png';

import STYLES from './sr-logo.scss';
import { cssModules } from 'bpk-react-utils';
const getClassName = cssModules(STYLES); // REGEX_REPLACED

const LONG_PRESS_DURATION = 3000;

class SRLogo extends Component {
  constructor(props) {
    super(props);

    this.state = {};
  }

  onMouseLeave = () => {};

  render() {
    const {
      noPadding,
      animated,
      small,
      className,
      alwaysCentered,
      light,
      ...rest
    } = this.props;
    const classNameFinal = [getClassName('sr-logo__container')];
    if (className) {
      classNameFinal.push(className);
    }
    if (alwaysCentered) {
      classNameFinal.push(getClassName('sr-logo__container--centered'));
    }
    if (animated) {
      classNameFinal.push(getClassName('sr-logo__container--animated'));
    }

    const baseTextClassNameFinal = [getClassName('sr-logo__logo-base')];
    if (light) {
      baseTextClassNameFinal.push(getClassName('sr-logo__logo-base--light'));
    }

    const largeTextClassNameFinal = [getClassName('sr-logo__logo-large')];
    if (small) {
      largeTextClassNameFinal.push(
        getClassName('sr-logo__logo-large--smaller'),
      );
    }
    if (noPadding) {
      classNameFinal.push(getClassName('sr-logo__container--no-padding'));
      largeTextClassNameFinal.push(
        getClassName('sr-logo__logo-large--no-padding'),
      );
      baseTextClassNameFinal.push(
        getClassName('sr-logo__logo-base--no-padding'),
      );
    }

    return (
      <div className={classNameFinal.join(' ')} {...rest}>
        <a
          role="button"
          aria-label="Main conference page"
          onMouseEnter={() => {
            this.setState({ hovering: true });
          }}
          onFocus={() => {
            this.setState({ hovering: true });
          }}
          onMouseLeave={() => {
            this.setState({ hovering: false });
          }}
          onBlur={() => {
            this.setState({ hovering: false });
          }}
          href="/"
          onMouseLeave={this.onMouseLeave}
        >
          <BpkImage
            src={Logo}
            className={largeTextClassNameFinal.join(' ')}
            width={1}
            height={1}
          />
        </a>
      </div>
    );
  }
}

SRLogo.propTypes = {
  className: PropTypes.string,
  noPadding: PropTypes.bool,
  animated: PropTypes.bool,
  small: PropTypes.bool,
  light: PropTypes.bool,
  alwaysCentered: PropTypes.bool,
};

SRLogo.defaultProps = {
  className: null,
  noPadding: false,
  animated: false,
  small: false,
  light: false,
  alwaysCentered: false,
};

export default SRLogo;
