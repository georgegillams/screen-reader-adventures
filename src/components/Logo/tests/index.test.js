import React from 'react';

import { render } from '@testing-library/react';

import Logo from '..';

describe('<Logo />', () => {
  it('Should render correctly', () => {
    const { container } = render(<Logo></Logo>);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly with custom className', () => {
    const { container } = render(<Logo className={'test-custom-classname'}></Logo>);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly pride', () => {
    const { container } = render(<Logo pride></Logo>);

    expect(container).toMatchSnapshot();
  });
  it('Should render correctly animated', () => {
    const { container } = render(<Logo animated></Logo>);

    expect(container).toMatchSnapshot();
  });
});
