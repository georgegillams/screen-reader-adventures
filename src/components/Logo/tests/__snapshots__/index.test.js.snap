// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Logo /> Should render correctly 1`] = `
<div>
  <div
    class="logo__container"
  >
    <a
      class="logo__a"
      href="/"
    >
      <h1
        aria-label="Home page"
        class="logo__heading"
        role="button"
      >
        LOGO
      </h1>
    </a>
  </div>
</div>
`;

exports[`<Logo /> Should render correctly animated 1`] = `
<div>
  <div
    class="logo__container"
  >
    <a
      class="logo__a"
      href="/"
    >
      <h1
        aria-label="Home page"
        class="logo__heading logo__subheading--animated"
        role="button"
      >
        LOGO
      </h1>
    </a>
  </div>
</div>
`;

exports[`<Logo /> Should render correctly pride 1`] = `
<div>
  <div
    class="logo__container"
  >
    <a
      class="logo__a"
      href="/"
    >
      <h1
        aria-label="Home page"
        class="logo__heading logo__heading--pride"
        role="button"
      >
        LOGO
      </h1>
    </a>
  </div>
</div>
`;

exports[`<Logo /> Should render correctly with custom className 1`] = `
<div>
  <div
    class="logo__container test-custom-classname"
  >
    <a
      class="logo__a"
      href="/"
    >
      <h1
        aria-label="Home page"
        class="logo__heading"
        role="button"
      >
        LOGO
      </h1>
    </a>
  </div>
</div>
`;
